src/global/project/mod.rs

```diff
@@ -80,31 +105,257 @@ impl Project {
+        let exposed = path
+            .file_stem()
+            .and_then(OsStr::to_str)
+            .ok_or_else(|| miette::miette!("Could not get file stem of {}", path.display()))
+            .and_then(ExposedKey::from_str)?;
+        let binary_path = Self::extract_bin_from_script(&path)?;
+
+        let binary = binary_path
+            .file_stem()
+            .and_then(OsStr::to_str)
+            .map(String::from)
+            .ok_or_else(|| miette::miette!("Could not get file stem of {}", path.display()))?;
+        let env_path = binary_path
+            .parent()
+            .ok_or_else(|| {
+                miette::miette!("binary_path '{}' has no parent", binary_path.display())
+            })?
+            .parent()
+            .ok_or_else(|| {
+                miette::miette!(
+                    "binary_path's parent '{}' has no parent",
+                    binary_path.display()
+                )
+            })?;
+        let env = env_path
+            .file_name()
+            .and_then(OsStr::to_str)
+            .ok_or_else(|| {
+                miette::miette!(
+                    "binary_path's grandparent '{}' has no file name",
+                    binary_path.display()
+                )
+            })
+            .and_then(|env| EnvironmentName::from_str(env).into_diagnostic())?;
+
+        let conda_meta = env_path.join("conda-meta");
+
+        let (platform, channel, package) = Self::package_from_conda_meta(&conda_meta, &binary)?;
+
+        Ok(ExposedData {
+            env,
+            platform,
+            channel,
+            package,
+            binary,
+            exposed,
+        })
+    }
+
+    fn package_from_conda_meta(
+        conda_meta: &Path,
+        binary: &str,
+    ) -> miette::Result<(Option<Platform>, PrioritizedChannel, PackageName)> {
+        for entry in std::fs::read_dir(conda_meta)
+            .into_diagnostic()
+            .wrap_err_with(|| format!("Couldn't read directory {}", conda_meta.display()))?
+        {
+            let path = entry
+                .into_diagnostic()
+                .wrap_err_with(|| {
+                    format!("Couldn't read file from directory {}", conda_meta.display())
+                })?
+                .path();
+            let channel_config = default_channel_config();
+            // Check if the entry is a file and has a .json extension
+            if path.is_file() && path.extension().and_then(|ext| ext.to_str()) == Some("json") {
+                let content = std::fs::read_to_string(&path).into_diagnostic()?;
+                let json: serde_json::Value = serde_json::from_str(&content)
+                    .into_diagnostic()
+                    .wrap_err_with(|| format!("Could not parse json from {}", path.display()))?;
+
+                // Check if the JSON contains the specified structure
+                if let Some(paths) = json.pointer("/paths_data/paths") {
```

Comment:
I'm thinking, maybe we can use rattler to do this heavy lifting?
There is `collect_from_prefix` in `rattler_conda_types::PrefixRecord`
that read the json and create `PrefixRecord`. From it we could extract Platform, Channel, package name and version

Reply:
Done!